royalties-registry/contracts/RoyaltiesRegistry.sol

```diff
@@ -6,36 +6,98 @@ pragma abicoder v2;
 import "@rarible/royalties/contracts/IRoyaltiesProvider.sol";
 import "@rarible/royalties/contracts/LibRoyaltiesV2.sol";
 import "@rarible/royalties/contracts/LibRoyaltiesV1.sol";
+import "@rarible/royalties/contracts/LibRoyalties2981.sol";
 import "@rarible/royalties/contracts/RoyaltiesV1.sol";
 import "@rarible/royalties/contracts/RoyaltiesV2.sol";
+import "@rarible/royalties/contracts/IERC2981.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 
 contract RoyaltiesRegistry is IRoyaltiesProvider, OwnableUpgradeable {
-
+    /// @dev deprecated
     event RoyaltiesSetForToken(address indexed token, uint indexed tokenId, LibPart.Part[] royalties);
+    /// @dev emitted when royalties set for token in 
     event RoyaltiesSetForContract(address indexed token, LibPart.Part[] royalties);
 
+    /// @dev struct to store royalties in royaltiesByToken
     struct RoyaltiesSet {
         bool initialized;
         LibPart.Part[] royalties;
     }
 
+    /// @dev deprecated
     mapping(bytes32 => RoyaltiesSet) public royaltiesByTokenAndTokenId;
+    /// @dev stores royalties for token contract, set in setRoyaltiesByToken() method
     mapping(address => RoyaltiesSet) public royaltiesByToken;
-    mapping(address => address) public royaltiesProviders;
+    /// @dev stores external provider and royalties type for token contract
+    mapping(address => uint) public royaltiesProviders;
+
+    /// @dev total amount or supported royalties types
+    // 1 - royaltiesByToken, 2 - v2, 3 - v1,
+    // 4 - external provider, 5 - EIP-2981
+    uint constant royaltiesTypesAmount = 5;
 
     function __RoyaltiesRegistry_init() external initializer {
         __Ownable_init_unchained();
     }
 
+    /// @dev sets external provider for token contract, and royalties type = 4
     function setProviderByToken(address token, address provider) external {
         checkOwner(token);
-        royaltiesProviders[token] = provider;
+        setRoyaltiesType(token, 4, provider);
+    }
+
+    /// @dev returns provider address for token contract from royaltiesProviders mapping
+    function getProvider(address token) public view returns(address) {
```

Comment:
I think need a little code formatting in this file like:  .....returns (address)